### PR TITLE
add lib for scheduler cluster CR

### PR DESCRIPTION
**Pull request by zhouying7780**
@xingxingxia @kasturinarra  could you please help review ? thanks

### CODE REVIEW

**Comment by zhouying7780** [comment]
@pruan-rht  Could you please help review ? thanks.

File: features/step_definitions/scheduler.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  else
+    raise "Could not get scheduler: #{name}"
+  end
+  patch_json = org_scheduler.to_json
+  _admin = admin
+  teardown_add {
+    opts = {resource: 'scheduler', resource_name: name, p: patch_json, type: 'merge' }
+    @result = _admin.cli_exec(:patch, **opts)
+    raise "Cannot restore scheduler: #{name}" unless @result[:success]
+    sleep 60
```
Comment (pruan-rht) [inline features/step_definitions/scheduler.rb]: @zhouying7780, can you please elaborate the reason for the 60 seconds hard-coded sleep?

Comment (xingxingxia) [reply features/step_definitions/scheduler.rb]: Agree, for any magic number, always encouraged to comment the reason

File: features/step_definitions/scheduler.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  }
+end
```
Comment (xingxingxia) [inline features/step_definitions/scheduler.rb]: Seems what this step does is common for many other CRs, could check if others defined. If not, seems better to make it common, not only to scheduler CR

File: features/step_definitions/scheduler.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+Given /^the #{QUOTED} scheduler CRD is restored after scenario$/ do |name|
```
Comment (xingxingxia) [inline features/step_definitions/scheduler.rb]: CR looks more accurate :), CR is Resource, CRD is Definition :)

File: features/step_definitions/scheduler.rb (hunk 2 above)
Comment (zhouying7780) [reply features/step_definitions/scheduler.rb]: @xingxingxia features/step_definitions/oauth.rb has defined the oauth CRD.

File: features/step_definitions/scheduler.rb (hunk 1 above)
Comment (zhouying7780) [reply features/step_definitions/scheduler.rb]: @pruan-rht @xingxingxia  I just copy the method from features/step_definitions/oauth.rb, it         has defined the 60 seconds sleep.

File: features/step_definitions/scheduler.rb (hunk 3 above)
Comment (zhouying7780) [reply features/step_definitions/scheduler.rb]: Have updated.

File: features/step_definitions/scheduler.rb (hunk 2 above)
Comment (xingxingxia) [reply features/step_definitions/scheduler.rb]: Just a suggestion to avoid same code multiple times. Fine to keep PR as is.

**Comment by xingxingxia** [comment]
lgtm with no more comment
@pruan-rht help checke / merge, thanks

**Comment by pruan-rht** [comment]
@zhouying7780 please rebase.  thanks!

**Comment by zhouying7780** [comment]
@pruan-rht  have rebased, please help review  and if no more need update please help merge,thanks.

**Comment by zhouying7780** [comment]
@liangxia   Could you please help review and merge , thanks.

File: features/step_definitions/scheduler.rb [hunk 4]
```diff
@@ -4,3 +4,23 @@
   @result = admin.cli_exec(:delete, object_type: 'priorityclasses', object_name_or_id: name)
   raise "priority classes were not deleted successfully!" unless @result[:success]
 end
+
+Given /^the #{QUOTED} scheduler CRD is restored after scenario$/ do |name|
+  ensure_admin_tagged
```
Comment (liangxia) [inline features/step_definitions/scheduler.rb]: Also need ensure_destructive_tagged as scheduler change can cause failures for other scenarios when running in parallel

**Comment by liangxia** [comment]
There are several unresolved comments need action.

**Comment by zhouying7780** [comment]
@liangxia   all comments have addressed ,  please help review again. thanks.

File: features/step_definitions/scheduler.rb (hunk 1 above)
Comment (zhouying7780) [reply features/step_definitions/scheduler.rb]: deleted , and retest , no issue.

**Pull request merged**